### ISSUE TITLE
Proximal map of inpainting operator

### ISSUE THREAD

**Issue by svvenkatakrishnan** [deepinv/deepinv#208]
For the simple forward model (binary selection) used for image inpainting,  Its way to avoid computing the proximal map using gradient descent and instead just use the closed form solution ?

**Comment by matthieutrs** [comment]
Hey @svvenkatakrishnan, in the case of [Inpainting](https://github.com/deepinv/deepinv/blob/main/deepinv/physics/inpainting.py#L5), the class inherits from the [DecomposablePhysics](https://github.com/deepinv/deepinv/blob/main/deepinv/physics/forward.py#L588) class, which has its own closed form implementation in the [prox_l2](https://github.com/deepinv/deepinv/blob/main/deepinv/physics/forward.py#L706) method, so gradient descent is not used! 

More generally, we've set the default computation of the prox as computed with an iterative method so that even if no implementation of the prox is provided by the user, a solution is returned. However, most Physics in the library overwrite this gradient descent implementation. As we've seen above, this is the case for all children of the DecomposablePhysics class, but not only - for example in some cases for [Downsampling](https://github.com/deepinv/deepinv/blob/main/deepinv/physics/blur.py#L266).

**Comment by svvenkatakrishnan** [comment]
Thank you for the detailed response @matthieutrs . Since it is using a closed form I was not sure what the "step size" parameter was doing. For example: 

Here is a code snippet I am trying for an inpainting problem with PnP-ADMM. 

physics = dinv.physics.Inpainting(
        mask=mask_tensor,
        tensor_size=y.shape,
        noise_model=dinv.physics.GaussianNoise(sigma=0),
        device=device
    )

denoiser=DRUNet(in_channels=1, out_channels=1, nc=[64, 128, 256, 512], nb=4, act_mode='R', downsample_mode='strideconv', upsample_mode='convtranspose', pretrained='download', train=False, device=device)

data_fidelity = L2()

prior = PnP(denoiser=denoiser)

params_algo = {"stepsize": _ss, "g_param": _gp, "lambda": 1.0} 

model = optim_builder(iteration="ADMM",
                          prior=prior,data_fidelity=data_fidelity,
                          early_stop=early_stop,
                          max_iter=max_iter,
                          verbose=verbose,
                          params_algo=params_algo
    )

 
    x_pnp_cnn, metrics = model(y, physics, x_gt=x, compute_metrics=True)

I have noticed that when I change the stepsize parameter the solution changes. Does the step size correspond to another variable in these cases ?

**Comment by matthieutrs** [comment]
Yes, this is indeed expected!

As you can see in ADMM the stepsize is used [here](https://github.com/deepinv/deepinv/blob/main/deepinv/optim/optim_iterators/admm.py#L91). More generally, just like in any other algorithm, a stepsize needs to be set. However, unlike what happens in optimization where the solution is independent of the stepsize, the solution of PnP algorithm depends on the external parameter chosen.

BTW, this is what makes DPIR quite interesting and powerful, stepsizes have been tuned with a specific decay for good performance :)

**Comment by svvenkatakrishnan** [comment]
@matthieutrs : Thank you.  However I think there is some terminology which I may be missing or is overloaded in the library. 

To solve the proximal map for the forward problem for the inpainting: ||y-Ax||_{2}^{2} + \gamma ||x - \tilde{x}||_{2}^{2}; it looks like the step size here is referring to \gamma. 

Instead, if this same prox is solved using say gradient descent, then stepsize could refer to something different in that context. This way one would need to set two parameters to solve the problem.  

Does that sound right ?

**Comment by svvenkatakrishnan** [comment]
Got it. It may not be urgent, but in a future release it would be good to expose the "inner iteration" parameters such as the step size, max inner iterations etc. as advanced settings that can be accessed using the params_algo.
